rust/src/main.rs

```diff
@@ -110,14 +117,14 @@ fn gen(args: Args) -> Result<()> {
         encode_sequence(
             &mut frequencies,
             length,
-            postings.iter().map(|p| p.get_tf()),
+            postings.iter().map(Posting::get_tf),
```

Comment:
This is an interesting little thing: you can do this thing because really `get_tf` acts as a free function with an argument that is a reference to `Posting`. This makes it easy to use methods like that.